pkg/operator/ceph/cluster/controller.go

```diff
@@ -360,29 +366,28 @@ func (c *ClusterController) configureLocalCephCluster(namespace, name string, cl
 				return false, nil
 			}
 
-			c.updateClusterStatus(clusterObj.Namespace, clusterObj.Name, cephv1.ClusterStateCreating, "")
+			config.ConditionExport(c.context, namespace, name,
+				cephv1.ConditionProgressing, v1.ConditionTrue, "ClusterProgressing", "Cluster is creating")
```

Comment:
This message shows after the operator is restarted, even if the cluster already exists. If the cluster already exists, can we instead show a message such as "Cluster is checking is updates are needed". To check for this condition, there are a few ways it could be done. Could you check the CR status for ceph health? If it's set to anything, that means the cluster was up previously. Something like this, assuming the CephStatus is populated:
```
if clusterObj.Status.CephStatus != "" {
  message = "Cluster is checking if updates are needed"
}
```

Reply:
Yeah. Something like this would be possible. I'll do this one. If I'm correct, then this progressing state would be shown only once and that is when the cluster is being created right? @travisn

Reply:
Yeah. I can implement this in the conditions.go file itself. Just need to check if the Cluster is on Ready state then Progressing will show this message

Reply:
Cluster is on Ready state in the sense if ClusterReady condition is True, then Progressing will be the mesage. @travisn

Reply:
Sounds good to check the ready condition.

Reply:
While doing this I observed that when the operator is restarted all the conditions are getting erased and starts from the beginning. I think I need to fix this also. Your thoughts on what this might be? @travisn

Reply:
The conditions are part of the `status` on the cluster CR. The status is periodically updated by the operator with the ceph status and its details. The ceph status isn't updated until after the first orchestration is completed, but perhaps some other status update is clearing the conditions if it's in the same status type.

Reply:
I am seeing that even if the operator restarted the ceph status and it's details are retained but somehow conditions are getting erased. I'll give this a  closer look tomorrow.

Reply:
Fixed after making some changes.